management/storageservice/client.go

```diff
@@ -128,14 +128,14 @@ func (self StorageServiceClient) GetBlobEndpoint(storageService *StorageService)
 	return "", errors.New(fmt.Sprintf(errBlobEndpointNotFound, storageService.ServiceName))
 }
 
-func (self *StorageServiceClient) createStorageServiceDeploymentConf(name, location string) StorageServiceDeployment {
+func (self *StorageServiceClient) createStorageServiceDeploymentConf(name, location string, accountType AccountType) StorageServiceDeployment {
```

Comment:
I guess I can merge like this and address these in a separate PR. General approach for 1:1 mapping between methods and REST API endpoints should be taking all parameters as a struct argument and all URL path components as arguments... 

e.g. create hosted service would definitely need name, so you'd have `name` in method signature but all other details would go into a `parameters CreateHostedServiceParameters` struct. Check C# SDK for examples. I think it is a great model and you can add new things without breaking the back compat on method signature.